apps/challenges/utils.py

```diff
@@ -20,3 +25,118 @@ def get_file_content(file_path, mode):
     if os.path.isfile(file_path):
         with open(file_path, mode) as file_content:
             return file_content.read()
+
+
+def convert_to_aws_ecr_compatible_format(string):
+    '''Make string compatible with AWS ECR repository naming
+
+    Arguments:
+        string {string} -- Desired ECR repository name
+
+    Returns:
+        string -- Valid ECR repository name
+    '''
+    return string.replace(" ", "-").lower()
+
+
+def get_or_create_ecr_repository(name, region_name='us-east-1'):
+    '''Get or create AWS ECR Repository
+
+    Arguments:
+        name {string} -- name of ECR repository
+
+    Keyword Arguments:
+        region_name {str} -- AWS region name (default: {'us-east-1'})
+
+    Returns:
+        tuple -- Contains repository dict and boolean field to represent whether ECR repository was created
+        Eg: (
+                {
+                    'repositoryArn': 'arn:aws:ecr:us-east-1:1234567890:repository/some-repository-name',
+                    'registryId': '1234567890',
+                    'repositoryName': 'some-repository-name',
+                    'repositoryUri': '1234567890.dkr.ecr.us-east-1.amazonaws.com/some-repository-name',
+                    'createdAt': datetime.datetime(2019, 2, 6, 9, 12, 5, tzinfo=tzlocal())
+                },
+                False
+            )
+    '''
+    AWS_ACCOUNT_ID = os.environ.get('AWS_ACCOUNT_ID')
+    repository, created = None, False
+    client = boto3.client('ecr', region_name=region_name)
+    try:
+        response = client.describe_repositories(
+            registryId=AWS_ACCOUNT_ID,
+            repositoryNames=[
+                name,
+            ]
+        )
+        repository = response['repositories'][0]
+    except ClientError as e:
+        if e.response['Error']['Code'] == 'RepositoryNotFoundException':
+            response = client.create_repository(repositoryName=name)
+            repository = response['repository']
+            created = True
```

Comment:
Please add an else block with logger.exception in case the create operation doesn't succeed.